src/libs/ReportUtils.ts

```diff
@@ -1939,20 +1933,19 @@ function getWorkspaceIcon(report: OnyxInputOrEntry<Report>, policy?: OnyxInputOr
     const workspaceName = getPolicyName(report, false, policy);
     const cacheKey = report?.policyID ?? workspaceName;
     const iconFromCache = workSpaceIconsCache.get(cacheKey);
-    const avatarURL = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]?.avatarURL;
-
-    const isSameAvatarURL = iconFromCache?.icon?.source === avatarURL;
-    const isDefaultWorkspaceAvatar = !avatarURL && typeof iconFromCache?.icon?.source !== 'string';
-    const hasWorkSpaceNameChanged = iconFromCache?.name !== workspaceName;
-    if (iconFromCache && (isSameAvatarURL || isDefaultWorkspaceAvatar) && !hasWorkSpaceNameChanged) {
-        return iconFromCache.icon;
-    }
     // disabling to protect against empty strings
     // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
     const policyAvatarURL = report?.policyAvatar || allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]?.avatarURL;
     // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
     const policyExpenseChatAvatarSource = policyAvatarURL || getDefaultWorkspaceAvatar(workspaceName);
 
+    const isSameAvatarURL = iconFromCache?.icon?.source === policyExpenseChatAvatarSource;
+    const hasWorkSpaceNameChanged = iconFromCache?.name !== workspaceName;
+
+    if (iconFromCache && (isSameAvatarURL || report?.policyAvatar === undefined) && !hasWorkSpaceNameChanged) {
```

Comment:
I doubt why we need condition `|| report?.policyAvatar === undefined`, isn't `isSameAvatarURL` enough?

Reply:
Let's say we already have a cache for workspace A custom avatar. If there is a report that has a missing `policyAvatar`, then `policyExpenseChatAvatarSource` will contain the default workspace avatar and override the cache.

Reply:
If the `policyAvatar` is missing then we are looking in the policy `avatarURL`, if that is missing too then we need to set the defaultAvatar and it should update the cache. isn't it? Why do we need to return old cache value?

Reply:
If the policyAvatar is an empty string, then yes, we should update the cache because that means the avatar is removed, but if it's missing (undefined), then it's just missing, not that the avatar is removed.

Reply:
Ok, I am not sure if that case will exist or not but I Got your point 👍